job_creator.php

```diff
@@ -13,23 +13,27 @@ public function getInstallerVersion(string $githubRepository, string $branch): s
         if (in_array($repo, NO_INSTALLER_REPOS)) {
             return '';
         }
-        // e.g. ['4', '11']
-        $portions = explode('.', $branch);
-        if (count($portions) == 1) {
-            return '4.x-dev';
+        // e.g. pulls/4.10/some-bugfix or pulls/4/some-feature
+        if (preg_match('#^pulls/([0-9\.]+)/#', $branch, $matches)) {
+            $branch = $matches[1];
```

Comment:
`$branch` is either a semantic version tag or `$myRef`, which according to a comment in `createJson` is the _target_ branch for PRs:
> // $myRef will either be a branch for push (i.e cron) and pull-request (target branch), or a semver tag

Is the intention here to explicitly allow CI to work correctly on the creative-commoners pr branches on push?
If so the comment I mentioned above should probably be updated.

Reply:
The comment about is correct and doesn't need to be changed

The method we're updating `getInstallerVersion()` is for selecting the version of `silverstripe/installer` to composer require.

The bit you've highlighted here is for creative-commoner pull-requests for lockstepped modules, though it has nothing to do with the $myRef variable

Reply:
> it has nothing to do with the `$myRef` variable

The `$myRef` variable holds the branch, which is passed in here as `$branch` - so it very much has to do with that variable. 

Perhaps there should just be an addition to the `// e.g. pulls/4.10/some-bugfix or pulls/4/some-feature` comment that says something like "for push events on the creative-commoners account" to make it clear what's happening here. It took me a bit to realise that's what the intention here probably is as opposed to checking the name of a branch on a PR.

Reply:
Updated